frost-core/src/verifying_key.rs

```diff
@@ -63,9 +68,15 @@ where
         //                 h * ( z * B - c * A - R) == 0
         //
         // where h is the cofactor
+        let mut R = signature.R;
+        let mut vk = self.element;
+        if <C>::is_need_tweaking() {
+            R = <C>::tweaked_R(&signature.R);
+            vk = <C>::tweaked_public_key(&self.element);
+        }
         let zB = C::Group::generator() * signature.z;
-        let cA = self.element * challenge.0;
-        let check = (zB - cA - signature.R) * C::Group::cofactor();
+        let cA = vk * challenge.0;
```

Comment:
OK so I think there's an issue here, vk might be the wrong tweaked public key  as `<C>::tweaked_public_key(&self.element);` on line 75 computes a tweaked public key point on the fly. But the tweaked public key point that Bitcoin might use is the one that is taken from the x coordinate only!